core/src/main/java/com/linecorp/armeria/common/stream/StreamMessageWriter.java

```diff
@@ -16,9 +16,12 @@
 
 package com.linecorp.armeria.common.stream;
 
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
 /**
- * A type which is both a {@link StreamMessage} and a {@link StreamWriter}. This type is mainly used by tests
- * which need to exercise both functionality.
+ * A type which is both a {@link StreamMessage} and a {@link StreamWriter}.
+ * {@link StreamMessageWriter} publishes the objects written via {@link StreamWriter#write(Object)}.
  */
-interface StreamMessageAndWriter<T> extends StreamMessage<T>, StreamWriter<T> {
+@UnstableApi
+public interface StreamMessageWriter<T> extends StreamMessage<T>, StreamWriter<T> {
```

Comment:
We neither provide an implemention that extends only `StreamWriter` nor public APIs that take `StreamWriter` as an input type. 
I’m not sure it is worth keeping both `StreamWriter` and `StreamMessageWriter`. 
It is likely that we can make `StreaWriter` extends `StreamMessage` and remove `StreamMessageWriter` for simplicity. 
@minwoox What do you think of that?

Reply:
That is a good suggestion and I agree with you on it. 👍 
What do you think of it? @line/dx

Reply:
If there aren't any APIs that should be on `StreamMessageWriter` only, then I think it's cleaner to use just `StreamWriter` 👍

Reply:
> It is likely that we can make StreaWriter extends StreamMessage and remove StreamMessageWriter for simplicity.

I also agree with this! looks simpler 👍

It's better to handle this in this PR or with another PR?

Reply:
We have some time before releasing the next version so it looks good to handle it in a separate issue.

Reply:
@trustin PTAL :bow:

Reply:
(If it's okay to handle with another PR, I'll create PR right after this PR is merged~!)

Reply:
SGTM

Reply:
✅ clean-up PR: https://github.com/line/armeria/pull/4718